platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/UiSettings.java

```diff
@@ -86,6 +86,19 @@ public void setCompassGravity(int gravity) {
     }
 
     /**
+     * Enables or disables fading of the compass when facing north.
+     * <p>
+     * By default this feature is enabled
+     * </p>
+     *
+     * @param compassFadeFacingNorth True to enable the fading animation; false to disable it
+     */
+    public void setCompassFadeFacingNorth(boolean compassFadeFacingNorth) {
+        compassSettings.setFadeFacingNorth(compassFadeFacingNorth);
+        mapView.setCompassFadeFacingNorth(compassFadeFacingNorth);
+    }
```

Comment:
I'm not sure if I completely follow.. End users use UiSettings as public API. UiSettings currently doesn't have any reference to CompassView itself so it's required to pass through MapView (This is following current implementation pattern but this will be removed in upcoming refactors!). In the UISettings method we are also setting the user  state of the compass in compassSettings. We use this class to restore state after an onsaveinstancestate. We could remove `compassSettings.setFadeFacingNorth(compassFadeFacingNorth);` but as a result we are not saving state.